operator/pkg/remote/kyma_synchronization_context.go

```diff
@@ -285,3 +289,68 @@ func GetRemoteObjectKey(kyma *operatorv1alpha1.Kyma) client.ObjectKey {
 	}
 	return client.ObjectKey{Namespace: namespace, Name: name}
 }
+
+type CatalogSettings struct {
+	Name      string
+	Namespace string
+}
+
+type CatalogEntry struct {
+	Defaults *unstructured.Unstructured `json:"defaults"`
+	Channel  v1alpha1.Channel           `json:"channel"`
+	Target   v1alpha1.Target            `json:"target"`
+	Version  string                     `json:"version"`
+}
+
+func (c *KymaSynchronizationContext) CreateOrUpdateModuleTemplateCatalog(
+	ctx context.Context, catalogSettings CatalogSettings, moduleTemplates *v1alpha1.ModuleTemplateList,
+) error {
+	catalog := &corev1.ConfigMap{}
+	catalog.SetName(catalogSettings.Name)
+	catalog.SetNamespace(catalogSettings.Namespace)
+
+	create := false
+	err := c.runtimeClient.Get(ctx, client.ObjectKeyFromObject(catalog), catalog)
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	create = k8serrors.IsNotFound(err)
+
+	if catalog.Data == nil {
+		catalog.Data = make(map[string]string)
+	}
+
+	for _, moduleTemplate := range moduleTemplates.Items {
+		moduleTemplate := &moduleTemplate
```

Comment:
with this way, it still copy a moduleTemplate, which is not necessary.

Reply:
https://github.com/kyma-project/kyma-operator/pull/103/commits/419dd905606805d9e26a67bfd52f7891d0d6678e#diff-bc1bb91f1f01617c2d3693314155dcfad6e5a612788e40ec921f8236d0a5a11cR323

I think you reviewed based on old state